datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachFlowConfig.java

```diff
@@ -61,12 +68,20 @@
 
             .from(SDX_ATTACH_NEW_CLUSTER_STATE).to(FINAL_STATE)
             .event(SDX_ATTACH_NEW_CLUSTER_SUCCESS_EVENT)
-            .failureState(SDX_ATTACH_NEW_CLUSTER_FAILED_STATE)
+            .failureState(SDX_DETACH_RECOVERY_STATE)
             .failureEvent(SDX_ATTACH_NEW_CLUSTER_FAILED_EVENT)
 
-            .from(SDX_ATTACH_NEW_CLUSTER_FAILED_STATE).to(SDX_DETACH_FAILED_STATE)
-            .event(SDX_DETACH_FAILED_EVENT).defaultFailureEvent()
+            .from(SDX_DETACH_RECOVERY_STATE).to(SDX_DETACH_FAILED_STATE)
+            .event(SDX_DETACH_RECOVERY_FAILURE_HANDLED_EVENT).defaultFailureEvent()
+
+            // Recovery flow.
+            .from(INIT_STATE).to(SDX_DETACH_RECOVERY_STATE)
+            .event(SDX_DETACH_RECOVERY_EVENT).noFailureEvent()
+
+            .from(SDX_DETACH_RECOVERY_STATE).to(FINAL_STATE)
+            .event(SDX_DETACH_RECOVERY_SUCCESS_EVENT).defaultFailureEvent()
```

Comment:
Is there a reason to have detach and reattach in the same flow?

Reply:
Not a huge reason aside from simplicity since the recovery process is identical to the automatic recovery done in the detach flow. It didn't make sense to me to make an entire new flow if the flow step would be the same on both.